yarn-project/aztec-nr/aztec/src/oracle/get_block_data.nr

```diff
@@ -0,0 +1,39 @@
+use dep::std::merkle::compute_merkle_root;
+use crate::{
+    abi::HistoricBlockData,
+    constants_gen::{
+        HISTORIC_BLOCK_DATA_LENGTH,
+        HISTORIC_BLOCKS_TREE_HEIGHT,
+    },
+    context::PrivateContext,
+    oracle::get_membership_witness::{
+        get_membership_witness,
+        MembershipWitness,
+    },
+};
+
+#[oracle(getBlockData)]
+fn get_block_data_oracle(_block_number: Field) -> [Field; HISTORIC_BLOCK_DATA_LENGTH] {}
+
+unconstrained pub fn get_block_data_internal(block_number: Field) -> HistoricBlockData {
+    let block_data = get_block_data_oracle(block_number);
+    HistoricBlockData::deserialize(block_data)
+}
+
+pub fn get_block_data(block_number: Field, context: PrivateContext) -> HistoricBlockData {
```

Comment:
Just me being annoying here. But should we wrap this in the context instead? 

Such that you could use it like: `context.get_block_data(block_number)` and that is it. Seems like more neat UX, it could still mostly live in here, just with that addition think it would be very nice to use.